### PR TITLE
Test for parquet pruning disabling

### PR DESCRIPTION
**Pull request by alamb**
# Which issue does this PR close?
Re https://github.com/apache/arrow-datafusion/issues/723

 # Rationale for this change
Tests are good!

# What changes are included in this PR?
Add a end to end test for the config flag added in https://github.com/apache/arrow-datafusion/pull/749

# Are there any user-facing changes?
No

### CODE REVIEW

**Comment by alamb** [comment]
FYI @lvheyang

File: datafusion/tests/parquet_pruning.rs [hunk 1]
```diff
@@ -136,6 +136,47 @@ async fn prune_date64() {
...
+    );
+
+    // same query, without pruning
+    let config = ExecutionConfig::new().with_parquet_pruning(false);
+
+    let output = ContextWithParquet::with_config(Scenario::Timestamps, config)
+        .await
+        .query(query)
+        .await;
+    println!("{}", output.description());
```
Comment (houqp) [inline datafusion/tests/parquet_pruning.rs:167]: is this print needed for the test?

Comment (alamb) [reply datafusion/tests/parquet_pruning.rs:167]: It is not needed for the test -- it is included in all of the tests in this module for debugging assistance (when the test fails a log of what is going on also is dumped). I can remove it if you prefer

Comment (houqp) [reply datafusion/tests/parquet_pruning.rs:167]: cool, just wanted to make sure it's intentional :)

**Review by houqp — APPROVED** [review]

**Comment by lvheyang** [comment]
This test looks great to me. I would add more test cases in the following PR. Thank you @alamb.

**Pull request merged**